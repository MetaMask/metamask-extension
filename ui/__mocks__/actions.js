const actions = require('../store/actions');
const {
  CONTRACT_ADDRESS_UNISWAP,
  CONTRACT_ADDRESS_SOURCIFY,
  CONTRACT_ADDRESS_FOUR_BYTE,
  TRANSACTION_DECODE_UNISWAP,
  TRANSACTION_DECODE_SOURCIFY,
  TRANSACTION_DECODE_FOUR_BYTE,
} = require('../../test/data/confirmations/transaction-decode');

const ERC20_TOKEN_1_MOCK = '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599'; // WBTC
const ERC20_TOKEN_2_MOCK = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'; // USDC

const TOKEN_DETAILS_MOCK = {
  [ERC20_TOKEN_1_MOCK]: {
    address: ERC20_TOKEN_1_MOCK,
    standard: 'ERC20',
    decimals: 8,
    name: 'Wrapped Bitcoin',
  },
  [ERC20_TOKEN_2_MOCK]: {
    address: ERC20_TOKEN_2_MOCK,
    standard: 'ERC20',
    decimals: 6,
    name: 'USD Coin',
  },
};

module.exports = {
  ...actions,
  getTokenStandardAndDetails: (address) => {
    if (!TOKEN_DETAILS_MOCK[address]) {
      console.log('Token not found:', address);

      return Promise.resolve({
        address,
        standard: 'ERC20',
        decimals: 18,
        name: 'Missing Mock',
      });
    }

    return Promise.resolve(TOKEN_DETAILS_MOCK[address]);
  },

  // eslint-disable-next-line no-empty-function
  trackMetaMetricsEvent: () => {},

  decodeTransactionData: async (request) => {
    const { contractAddress } = request;

    if (contractAddress === CONTRACT_ADDRESS_UNISWAP) {
      return TRANSACTION_DECODE_UNISWAP;
    } else if (contractAddress === CONTRACT_ADDRESS_SOURCIFY) {
      return TRANSACTION_DECODE_SOURCIFY;
    } else if (contractAddress === CONTRACT_ADDRESS_FOUR_BYTE) {
      return TRANSACTION_DECODE_FOUR_BYTE;
    }

    return undefined;
  },
};
