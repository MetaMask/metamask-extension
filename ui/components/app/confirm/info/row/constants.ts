export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';

export enum RowAlertKey {
  EstimatedFee = 'estimatedFee',
  SigningInWith = 'signingInWith',
  RequestFrom = 'requestFrom',
  Resimulation = 'resimulation',
  Speed = 'speed',
}

export enum AlertActionKey {
  Buy = 'buy',
  ShowAdvancedGasFeeModal = 'showAdvancedGasModal',
  ShowGasFeeModal = 'showGasFeeModal',
}
