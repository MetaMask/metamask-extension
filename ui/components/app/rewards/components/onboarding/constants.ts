export const REWARDS_ONBOARD_OPTIN_LEGAL_LEARN_MORE_URL =
  'https://support.metamask.io/manage-crypto/metamask-rewards';

export const REWARDS_ONBOARD_TERMS_URL = 'https://go.metamask.io/rewards-terms';
