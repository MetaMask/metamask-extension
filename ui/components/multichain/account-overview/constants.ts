export const FUND_SLIDE = {
  id: 'fund',
  title: 'slideFundWalletTitle',
  description: 'slideFundWalletDescription',
  image: './images/slide-fund-icon.svg',
  href: 'https://portfolio.metamask.io/buy/build-quote',
};

///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
export const BRIDGE_SLIDE = {
  id: 'bridge',
  title: 'slideBridgeTitle',
  description: 'slideBridgeDescription',
  image: './images/slide-bridge-icon.svg',
};
///: END:ONLY_INCLUDE_IF

export const CARD_SLIDE = {
  id: 'card',
  title: 'slideDebitCardTitle',
  description: 'slideDebitCardDescription',
  image: './images/slide-card-icon.svg',
  href: 'https://portfolio.metamask.io/card',
};

export const CASH_SLIDE = {
  id: 'cash',
  title: 'slideCashOutTitle',
  description: 'slideCashOutDescription',
  image: './images/slide-sell-icon.svg',
  href: 'https://portfolio.metamask.io/sell',
};

export const ZERO_BALANCE = '0x00';
