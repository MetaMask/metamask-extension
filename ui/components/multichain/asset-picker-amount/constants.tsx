export const LARGE_SYMBOL_LENGTH = 7;
