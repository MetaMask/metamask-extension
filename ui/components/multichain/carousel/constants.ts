export const MARGIN_VALUES = {
  SLIDE_BOTTOM: '40px',
  CONTAINER_SIDE: '16px',
  SLIDE_GAP: '2%',
  ZERO: '0',
};

export const WIDTH_VALUES = {
  FULL_WIDTH: '100%',
  STANDARD_SLIDE: '98%',
  FIRST_SLIDE: '94%',
};

export const BANNER_STYLES = {
  HEIGHT: '59px',
};

export const MAX_SLIDES = 5;
