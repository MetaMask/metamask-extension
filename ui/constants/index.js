export const ETH_DEFAULT_DECIMALS = 8;
