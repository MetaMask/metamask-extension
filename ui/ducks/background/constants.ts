import type {
  BackgroundStateProxy,
  MemStoreControllersComposedState,
} from '../../../shared/types/background';

export const initialBackgroundState: Omit<
  BackgroundStateProxy,
  keyof MemStoreControllersComposedState
> &
  Partial<{
    [ControllerName in keyof MemStoreControllersComposedState]: Partial<
      MemStoreControllersComposedState[ControllerName]
    >;
  }> = {
  isInitialized: false,
} as const;
