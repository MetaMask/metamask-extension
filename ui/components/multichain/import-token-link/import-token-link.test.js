import React from 'react';
import configureMockStore from 'redux-mock-store';
import { fireEvent, screen } from '@testing-library/react';
import { detectTokens } from '../../../store/actions';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import { mockNetworkState } from '../../../../test/stub/networks';
import ImportControl from '../../app/assets/asset-list/import-control';
import { ImportTokenLink } from '.';

const mockPushHistory = jest.fn();

jest.mock('react-router-dom', () => {
  const original = jest.requireActual('react-router-dom');
  return {
    ...original,
    useLocation: jest.fn(() => ({ search: '' })),
    useHistory: () => ({
      push: mockPushHistory,
    }),
  };
});

jest.mock('../../../store/actions.ts', () => ({
  detectTokens: jest.fn().mockImplementation(() => ({ type: 'DETECT_TOKENS' })),
  showImportTokensModal: jest
    .fn()
    .mockImplementation(() => ({ type: 'UI_IMPORT_TOKENS_POPOVER_OPEN' })),
}));

describe('Import Token Link', () => {
  it('should match snapshot for goerli chainId', () => {
    const mockState = {
      metamask: {
        ...mockNetworkState({ chainId: CHAIN_IDS.GOERLI }),
      },
    };

    const store = configureMockStore()(mockState);

    const { container } = renderWithProvider(<ImportTokenLink />, store);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for mainnet chainId', () => {
    const mockState = {
      metamask: {
        ...mockNetworkState({ chainId: CHAIN_IDS.MAINNET }),
      },
    };

    const store = configureMockStore()(mockState);

    const { container } = renderWithProvider(<ImportTokenLink />, store);

    expect(container).toMatchSnapshot();
  });

  it('should detectTokens when clicking refresh', () => {
    const mockState = {
      metamask: {
        ...mockNetworkState({ chainId: CHAIN_IDS.GOERLI }),
      },
    };

    const store = configureMockStore()(mockState);

    renderWithProvider(<ImportTokenLink />, store); // should this be RefreshTokenLink?

    const refreshList = screen.getByTestId('refresh-list-button');
    fireEvent.click(refreshList);

    expect(detectTokens).toHaveBeenCalled();
  });

  it('should push import token route', () => {
    const mockState = {
      metamask: {
        ...mockNetworkState({ chainId: CHAIN_IDS.GOERLI }),
      },
    };

    const store = configureMockStore()(mockState);

    renderWithProvider(<ImportControl />, store);

    const importToken = screen.getByTestId('import-token-button');
    fireEvent.click(importToken);

    expect(screen.getByText('Import')).toBeInTheDocument();
  });
});
