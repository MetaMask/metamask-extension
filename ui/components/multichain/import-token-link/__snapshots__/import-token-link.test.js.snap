// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Token Link should match snapshot for goerli chainId 1`] = `
<div>
  <div
    class="mm-box multichain-import-token-link"
  >
    <div
      class="mm-box mm-box--padding-top-2 mm-box--display-flex mm-box--align-items-center"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-link mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
        data-testid="refresh-list-button"
      >
        <span
          class="mm-box mm-icon mm-icon--size-sm mm-box--margin-inline-end-1 mm-box--display-inline-block mm-box--color-inherit"
          style="mask-image: url('./images/icons/refresh.svg');"
        />
        Refresh list
      </button>
    </div>
  </div>
</div>
`;

exports[`Import Token Link should match snapshot for mainnet chainId 1`] = `
<div>
  <div
    class="mm-box multichain-import-token-link"
  >
    <div
      class="mm-box mm-box--padding-top-2 mm-box--display-flex mm-box--align-items-center"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-link mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
        data-testid="refresh-list-button"
      >
        <span
          class="mm-box mm-icon mm-icon--size-sm mm-box--margin-inline-end-1 mm-box--display-inline-block mm-box--color-inherit"
          style="mask-image: url('./images/icons/refresh.svg');"
        />
        Refresh list
      </button>
    </div>
  </div>
</div>
`;
